### PR TITLE
Draw edge feature

### PR DESCRIPTION
**Pull request by EugenKoulik**

### CODE REVIEW

File: CraphModel/Edge.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    class Edge
+    {
+        public int Id { get; set; }
+
+        public int WeightEdge { get; set; }
+
+        public int ConnectableEdge { get; set; }
+
+    }
+}
```
Comment (Maksim-Trolina) [inline CraphModel/Edge.cs]: There is already a Vertex class that performs this function

File: GraphModelDraw/EdgeDraw.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    public class EdgeDraw
+    {
+        public readonly BrushColor BrushEdge;
+
+        public readonly int WeightEdge;
+
+        public readonly int Id;
+
+        public readonly int ConnectabelVertex;
+
```
Comment (Maksim-Trolina) [inline GraphModelDraw/EdgeDraw.cs]: There is no need to describe the weight in the model drawing.It is necessary to create a field for the Id of the vertex, from which it comes and to which it is directed

File: Forms/VertexClick.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿using System;
+
+public class Class1
+{
+	public Class1()
+	{
+	}
+}
```
Comment (Maksim-Trolina) [inline Forms/VertexClick.cs]: I think something is superfluous

File: Forms/DrawForm/DrawForm.cs [hunk 4]
```diff
@@ -57,15 +81,35 @@ private void MouseClickDrawForm(object sender, MouseEventArgs e)
...
+                    {
+                        EdgeDraw edgeDraw = new EdgeDraw(BrushColor.Black, 0, startVertexId, endVertexId);
+
+                        edgeDraws.Add(edgeDraw);
+                        Refresh();
+
+                        startVertexId = -1;
+                        endVertexId = -1;
+                    }
+
```
Comment (Maksim-Trolina) [inline Forms/DrawForm/DrawForm.cs]: Too complicated logic, separate class needed

File: Forms/DrawForm/DrawForm.cs [hunk 5]
```diff
@@ -74,13 +118,47 @@ protected override void OnPaint(PaintEventArgs e)
...
+                    if (edge.Id == vertex.Id)
+                    {
+                        startX = vertex.X + (int)VertexParameters.Radius;
+                        startY = vertex.Y + (int)VertexParameters.Radius;
+
+                    }
+                    else if (edge.ConnectabelVertex == vertex.Id)
+                    {
+                        endX = vertex.X + (int)VertexParameters.Radius;
+                        endY = vertex.Y + (int)VertexParameters.Radius;
```
Comment (Maksim-Trolina) [inline Forms/DrawForm/DrawForm.cs]: Too complicated logic, separate class needed

File: Forms/DrawForm/DrawForm.cs (hunk 4 above)
Comment (EugenKoulik) [reply Forms/DrawForm/DrawForm.cs]: fuck you, dominik

File: Forms/DrawForm/DrawForm.cs [hunk 6]
```diff
@@ -19,7 +25,16 @@ public partial class DrawForm : Form
 
         private Font vertexTextFont;
 
-        public DrawForm(List<VertexDraw> vertexDraws)
+        private VertexClick vertexClick;
+
+        private int startVertexId;
+
+        private int endVertexId;
```
Comment (Maksim-Trolina) [inline Forms/DrawForm/DrawForm.cs]: Why store these fields in the form?

File: Forms/DrawForm/DrawForm.cs [hunk 7]
```diff
@@ -74,13 +111,29 @@ protected override void OnPaint(PaintEventArgs e)
 
             graphics.TextRenderingHint = System.Drawing.Text.TextRenderingHint.AntiAlias;
 
+
+            foreach (var edge in edgeDraws)
+            {
+                Point p1 = new Point();
+                Point p2 = new Point();
```
Comment (Maksim-Trolina) [inline Forms/DrawForm/DrawForm.cs]: Why initialize new objects every time?

**Review by Maksim-Trolina — APPROVED** [review]

**Review by dnbss — APPROVED** [review]

**Pull request merged**